File: src/SSHServer/Ciphers/ICipher.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SSHServer.Ciphers
{
    public interface ICipher : IAlgorithm
    {
        uint BlockSize { get; }
        uint KeySize { get; }
        byte[] Encrypt(byte[] data);
        byte[] Decrypt(byte[] data);
        void SetKey(byte[] key, byte[] iv);
    }
}

File: SSHServerNodeJS/SSHServerNodeJS/Ciphers/ICipher.ts
import { IAlgorithm } from "../IAlgorithm";

export interface ICipher extends IAlgorithm {
    getBlockSize(): number;
    getKeySize(): number;
    encrypt(data: Buffer): Buffer;
    decrypt(data: Buffer): Buffer;
    setKey(key: Buffer, iv: Buffer): void;
}
